"unrecognized command line option ‘-march=native’; did you mean ‘-mcpu=native’?" on ppc64el

From a PPA build of the master branch on Ubuntu 20.04 (https://launchpadlibrarian.net/526787899/buildlog_ubuntu-focal-ppc64el.givaro_4.1.1.169.gcf0f715e~ubuntu20.04.1_BUILDING.txt.gz):

```
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -Wdate-time -D_FORTIFY_SOURCE=2 -O2 -march=native -Wall -DNDEBUG -UDEBUG -g -O3 "-fdebug-prefix-map=/<<PKGBUILDDIR>>=." -fstack-protector-strong -Wformat -Werror=format-security -c givinit.C  -fPIC -DPIC -o .libs/givinit.o
g++: error: unrecognized command line option ‘-march=native’; did you mean ‘-mcpu=native’?
make[5]: *** [Makefile:475: giverror.lo] Error 1
make[5]: *** Waiting for unfinished jobs....
g++: error: unrecognized command line option ‘-march=native’; did you mean ‘-mcpu=native’?
g++: error: unrecognized command line option ‘-march=native’; did you mean ‘-mcpu=native’?
g++: error: unrecognized command line option ‘-march=native’; did you mean ‘-mcpu=native’?
make[5]: *** [Makefile:475: givbasictype.lo] Error 1
```

The same thing happened in other Ubuntu releases as well:
* 18.04 (https://code.launchpad.net/~macaulay2/+archive/ubuntu/macaulay2/+build/21128460/+files/buildlog_ubuntu-bionic-ppc64el.givaro_4.1.1.169.gcf0f715e~ubuntu18.04.1_BUILDING.txt.gz)
* 20.10 (https://code.launchpad.net/~macaulay2/+archive/ubuntu/macaulay2/+build/21128449/+files/buildlog_ubuntu-groovy-ppc64el.givaro_4.1.1.169.gcf0f715e~ubuntu20.10.1_BUILDING.txt.gz)
* 21.04 (https://code.launchpad.net/~macaulay2/+archive/ubuntu/macaulay2/+build/21128454/+files/buildlog_ubuntu-hirsute-ppc64el.givaro_4.1.1.169.gcf0f715e~ubuntu21.04.1_BUILDING.txt.gz)

--- a/configure.ac
+++ b/configure.ac
@@ -65,7 +65,7 @@ SET_FLAGS
 
 # Append -march=native or -mcpu=native (if recognized by the compiler) to
 # OPTIM_FLAGS if not present in CXXFLAGS and not cross-compiling and
-# --without-marchnative is not set
+# --without-archnative is not set
 ARCH_FLAGS
 
 # Append -mfpmath=sse to OPTIM_FLAGS on i386 and i686 architecture with SSE

--- a/configure.ac
+++ b/configure.ac
@@ -63,8 +63,9 @@ AC_INLINE # DON'T TOUCH
 # Set OPTIM_FLAGS, DEBUG_FLAGS depending on compiler and command line arguments
 SET_FLAGS
 
-# Append -march=native to OPTIM_FLAGS if not present in CXXFLAGS and
-# not cross-compiling and --no-marchnative is not set
+# Append -march=native or -mcpu=native (if recognized by the compiler) to
+# OPTIM_FLAGS if not present in CXXFLAGS and not cross-compiling and
+# --without-marchnative is not set
 ARCH_FLAGS
 
 # Append -mfpmath=sse to OPTIM_FLAGS on i386 and i686 architecture with SSE

--- a/macros/common.m4
+++ b/macros/common.m4
@@ -67,23 +67,52 @@ AC_DEFUN([SET_FLAGS],[
           ])
     ])
 
+AC_DEFUN([AX_CHECK_COMPILE_FLAG],
+[dnl
+AS_VAR_PUSHDEF([CACHEVAR],[ax_cv_check_[]_AC_LANG_ABBREV[]flags_$4_$1])dnl
+AC_CACHE_CHECK([whether _AC_LANG compiler accepts $1], CACHEVAR, [
+  ax_check_save_flags=$[]_AC_LANG_PREFIX[]FLAGS
+  _AC_LANG_PREFIX[]FLAGS="$[]_AC_LANG_PREFIX[]FLAGS $4 $1"
+  AC_COMPILE_IFELSE([AC_LANG_SOURCE(m4_default([$5],[AC_LANG_PROGRAM()]))],
+    [AS_VAR_SET(CACHEVAR,[yes])],
+    [AS_VAR_SET(CACHEVAR,[no])])
+  _AC_LANG_PREFIX[]FLAGS=$ax_check_save_flags])
+AS_VAR_IF(CACHEVAR,yes,
+  [m4_default([$2], :)],
+  [m4_default([$3], :)])
+AS_VAR_POPDEF([CACHEVAR])dnl
+])dnl AX_CHECK_COMPILE_FLAGS
 
 
-dnl Append -march=native to OPTIM_FLAGS if not present in CXXFLAGS and
-dnl target==host and --no-marchnative is not set
+dnl Append -march=native or -mcpu=native (if recognized by the compiler) to
+dnl OPTIM_FLAGS if not present in CXXFLAGS and not cross-compiling and
+dnl --without-marchnative is not set
 AC_DEFUN([ARCH_FLAGS],[
     AC_ARG_WITH(archnative, [AC_HELP_STRING([--without-archnative],
-        [do not use -march=native (default is to use it if -march is not present in CXXFLAGS)])])
+        [do not use -march=native or -mcpu=native (default is to use it if not already present in CXXFLAGS)])])
 
-    AS_CASE([$CXXFLAGS],
-            [*-march=*], [], # do nothing if already set in CXXFLAGS
-            [AS_IF([test "x${with_archnative}" == "xno"],
-                [], # do nothing if option is set to no
-                [AS_IF([test "${host}" != "${build}" -o "${host}" != "${target}"],
-                    [AC_MSG_NOTICE("For efficiency you may want to add a '-march=...' flag in CXXFLAGS")],
-                    [AC_MSG_NOTICE("Adding '-march=native' to OPTIM_FLAGS")
-                    OPTIM_FLAGS+=" -march=native"])])])
-    ])
+    AX_CHECK_COMPILE_FLAG([-march=native], [
+        AS_CASE([$CXXFLAGS],
+                [*-march=*], [], # do nothing if already set in CXXFLAGS
+                [AS_IF([test "x${with_archnative}" == "xno"],
+                    [], # do nothing if option is set to no
+                    [AS_IF([test "${host}" != "${build}" -o "${host}" != "${target}"],
+                        [AC_MSG_NOTICE("For efficiency you may want to add a '-march=...' flag in CXXFLAGS")],
+                        [AC_MSG_NOTICE("Adding '-march=native' to OPTIM_FLAGS")
+                        OPTIM_FLAGS+=" -march=native"])])])
+    ], [
+        AX_CHECK_COMPILE_FLAG([-mcpu=native], [
+            AS_CASE([$CXXFLAGS],
+                    [*-cpu=*], [], # do nothing if already set in CXXFLAGS
+                    [AS_IF([test "x${with_archnative}" == "xno"],
+                        [], # do nothing if option is set to no
+                        [AS_IF([test "${host}" != "${build}" -o "${host}" != "${target}"],
+                            [AC_MSG_NOTICE("For efficiency you may want to add a '-cpu=...' flag in CXXFLAGS")],
+                            [AC_MSG_NOTICE("Adding '-mcpu=native' to OPTIM_FLAGS")
+                            OPTIM_FLAGS+=" -mcpu=native"])])])
+        ], [], [], [])
+    ], [], [])
+])
 
 dnl Append -mfpmath=sse to OPTIM_FLAGS on i386 and i686 architecture with SSE
 AC_DEFUN([FPMATH_FLAGS],[

--- a/macros/common.m4
+++ b/macros/common.m4
@@ -86,7 +86,7 @@ AS_VAR_POPDEF([CACHEVAR])dnl
 
 dnl Append -march=native or -mcpu=native (if recognized by the compiler) to
 dnl OPTIM_FLAGS if not present in CXXFLAGS and not cross-compiling and
-dnl --without-marchnative is not set
+dnl --without-archnative is not set
 AC_DEFUN([ARCH_FLAGS],[
     AC_ARG_WITH(archnative, [AC_HELP_STRING([--without-archnative],
         [do not use -march=native or -mcpu=native (default is to use it if not already present in CXXFLAGS)])])

